### PR TITLE
Upgrade Appium

### PR DESCRIPTION
**Pull request by MadeinFrance**
- Add XCUITest as DEV dependency
- Add Typescript

<hr>
Blocking update:

- `"webdriverio": "^7.30.1"` to `v8`

Reason:

Client/driver after the remote connection is `undefined`

 `TypeError: Cannot read properties of undefined (reading '$')`

      48 |
      49 |   test('First test', async function () {
    > 50 |     const loginStatusText = await client.$('~login-status');

### CODE REVIEW

**Comment by MadeinFrance** [comment]
Hi @kelset anything I can do to speed up this PR?

File: package.json [hunk 1]
```diff
@@ -4,31 +4,34 @@
...
+    "appium": "^2.0.0-beta.53",
+    "appium-xcuitest-driver": "^4.16.12",
     "babel-jest": "^26.6.3",
     "eslint": "^7.32.0",
     "jest": "^26.6.3",
-    "webdriverio": "^7.21.0",
     "metro-react-native-babel-preset": "^0.70.3",
-    "react-test-renderer": "18.0.0"
+    "react-test-renderer": "18.0.0",
+    "webdriverio": "^7.30.1"
```
Comment (MadeinFrance) [inline package.json:34]: To be updated in another PR

**Comment by kelset** [comment]
hey @MadeinFrance, sorry, I have a lot on my plate atm - I'll try to get back to this by end of the week

File: package.json (hunk 1 above)
Comment (MadeinFrance) [reply package.json:34]: [branch](https://github.com/MadeinFrance/react-native-e2e-jest-appium-webdriverio/commit/d0faecfa1c115605dd649d128f3475081a9fda57#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R43)

Issue:
```
node_modules/webdriverio/build/index.js:1
    ({"Object.<anonymous>":function(module,exports,require,__dirname,__filename,global,jest){import logger from '@wdio/logger';

SyntaxError: Cannot use import statement outside a module
```
